sdk/azcore/internal/pollers/util.go

```diff
@@ -53,6 +53,17 @@ func IsValidURL(s string) bool {
 }
 
 const idSeparator = ";"
+const emptyTypeName = "-empty-"
+
+// PollerTypeName returns the type name to use when constructing the poller ID.
+// It handles the empty type struct{} with a special name instead of the empty string.
+func PollerTypeName[T any]() string {
+	n := shared.TypeOfT[T]().Name()
+	if n == "" {
+		n = emptyTypeName
```

Comment:
We could also just disallow specifying an empty type (e.g. `struct{}`) instead, although this was useful for tests where the type is irrelevant.  If people think it's safer to just disallow instead of handling empty types specially I can make that change.

Reply:
I went ahead and made this change as it's representative of real-world scenarios.  We can always relax it in a non-breaking way in the future if needed.